### ISSUE TITLE
Invalid non-string/buffer chunk

### ISSUE THREAD

**Issue by kjdev** [mscgenjs/mscgenjs-cli#21]
Invalid when command is executed

```
$ PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1 npm install --global mscgenjs-cli
$ mscgenjs -i example.xu -T png
Invalid non-string/buffer chunk
```

**Comment by sverweij** [comment]
Hi @kjdev - why the `PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1` ? 

`mscgenjs-cli` uses puppeteer to render the sequence charts, and puppeteer needs chromium. If chromium isn't there I'd expect puppeteer to fail in unexpected ways - and that seems to be exactly what's happening.

I've tried it _with_ and _without_ the environment variable, and _without_ the environment variable things work fine (I did need to do an `npm uninstall --global mscgenjs-cli` before the second `npm i --global mscgenjs-cli`, because otherwise npm would keep using the chromium-less puppeteer it cached earlier...). 

Could you retry without the `PUPPETEER_SKIP_CHROMIUM_DOWNLOAD`?

**Comment by kjdev** [comment]
Added the PUPPETEER_SKIP_CHROMIUM_DOWNLOAD variable to suppress puppeteer errors when installing mscgenjs-cli.

``` sh
$ npm i -g mscgenjs-cli
/usr/bin/mscgen_js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgenjs -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgen.js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
npm WARN lifecycle puppeteer@1.1.1~install: cannot run in wd %s %s (wd=%s) puppeteer@1.1.1 node install.js /mscgenjs-cli/node_modules/puppeteer
+ mscgenjs-cli@2.0.5
```

Run as if installing puppeteer.

``` sh
$ npm i -g --unsafe-perm=true mscgenjs-cli
/usr/bin/mscgenjs -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgen_js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgen.js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
> puppeteer@1.1.1 install /usr/lib/node_modules/mscgenjs-cli/node_modules/puppeteer
> node install.js
Downloading Chromium r536395 - 96.2 Mb [====================] 100% 0.0s
Chromium downloaded to /usr/lib/node_modules/mscgenjs-cli/node_modules/puppeteer/.local-chromium/linux-536395
+ mscgenjs-cli@2.0.5
```

Even this was an error.

``` sh
$ mscgenjs -i example.xu -T png
Invalid non-string/buffer chunk
```

In my execution environment (Fedora 27 on docker), operation of headless chrome from puppeteer was not done normally.

It now works by manually installing Chrome and changing puppeteer options.

``` sh
$ dnf -y install https://dl.google.com/linux/direct/google-chrome-stable_current_x86_64.rpm
$ export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
$ npm install -g mscgenjs-cli
/usr/bin/mscgen_js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgen.js -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
/usr/bin/mscgenjs -> /usr/lib/node_modules/mscgenjs-cli/bin/mscgen_js
> puppeteer@1.1.1 install /usr/lib/node_modules/mscgenjs-cli/node_modules/puppeteer
> node install.js
**INFO** Skipping Chromium download. "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" environment variable was found.
+ mscgenjs-cli@2.0.5
```

Updated `/usr/lib/node_modules/mscgenjs-cli/dist/actions/render.js`.

``` js
browser = yield puppeteer.launch({
    headless: true,
    executablePath: '/usr/bin/google-chrome',
    args: ['--no-sandbox', '--disable-setuid-sandbox']
});
```

**Comment by sverweij** [comment]
@kjdev clear - so it's a problem with puppeteer installation. I can't help there  - but what I _can_ do is add a configuration option (e.g. via command line or an environment variable) to mscgenjs-cli so you can set the executablePath (and args, I guess) without having to tinker in the dist each time a new version of mscgenjs-cli is released.

https://github.com/orgs/mscgenjs/projects/1#card-8142591

**Comment by sverweij** [comment]
@kjdev thanks to your research and very clear explanation  (thank you!!) I was able to make a simple addition to mscgenjs-cli that makes puppeteer launch options configurable - see PR #23 for details. 

I've published it on npm on the beta channel as `mscgenjs-cli@2.1.0-beta-1`- so if you want to you can check if it works for you (let me know if it does (or doesn't)!)

I'll be polishing the documentation (and possibly tightening the validation on the configuration file a bit) before releasing.

**Comment by kjdev** [comment]
@sverweij great!

It worked with setting of `mscgenjs-cli@2.1.0-beta-1` and puppeteer-options.

```
$ npm install -g mscgenjs-cli@2.1.0-beta-1
$ cat > pupetter-options.json <<EOF
{
    "headless": true,
    "executablePath": "/usr/bin/google-chrome",
    "args": ["--no-sandbox", "--disable-setuid-sandbox"]
}
EOF
$ mscgenjs coolchart.msc -T png --puppeteer-options puppeteer-options.json
$ : created coolchart.png
```

**Comment by sverweij** [comment]
@kjdev I've released mscgenjs-cli@2.1.0 - it contains this fix, a stricter control over what puppeteer options can be passed and an improvement to the error reporting: instead of `non-string/ buffer chunk` it now properly reports the root cause.
